src/ConnectedMode.UnitTests/Migration/FileProviderTests.cs

```diff
@@ -31,11 +40,115 @@ public class MSBuildFileProviderTests
         public void MefCtor_CheckIsExported()
         {
             MefTestHelpers.CheckTypeCanBeImported<MSBuildFileProvider, IFileProvider>(
-                MefTestHelpers.CreateExport<ILogger>());
+                MefTestHelpers.CreateExport<SVsServiceProvider>(),
+                MefTestHelpers.CreateExport<ILogger>(),
+                MefTestHelpers.CreateExport<IThreadHandling>());
         }
 
         [TestMethod]
         public void MefCtor_CheckTypeIsNonShared()
             => MefTestHelpers.CheckIsNonSharedMefComponent<MSBuildFileProvider>();
+
+        [TestMethod]
+        public async Task GetFiles_NoSolutionFolder_ReturnsEmptyList()
+        {
+            var solution = CreateIVsSolution(null /* no solution folder */);
+            var serviceProvider = CreateServiceProviderWithSolution(solution.Object);
+
+            var fileSystem = new Mock<IFileSystem>();
+
+            var testSubject = CreateTestSubject(serviceProvider.Object, fileSystem.Object);
+
+            var actual = await testSubject.GetFilesAsync(CancellationToken.None);
+
+            actual.Should().BeEmpty();
+            fileSystem.Invocations.Should().HaveCount(0);
+        }
+
+        [TestMethod]
+        public async Task GetFiles_HasSolutionFolder_ExpectedFilesReturned()
+        {
+            var solution = CreateIVsSolution("root dir");
+            var serviceProvider = CreateServiceProviderWithSolution(solution.Object);
+
+            var filesToReturn = new string[] { "file1", "file2", "file3"};
+            var fileSystem = CreateFileSystem(filesToReturn);
+
+            var testSubject = CreateTestSubject(serviceProvider.Object, fileSystem.Object);
+
+            var actual = await testSubject.GetFilesAsync(CancellationToken.None);
+
+            actual.Should().BeEquivalentTo(filesToReturn);
+            solution.VerifyAll();
+
+            fileSystem.Verify(x => x.Directory.GetFiles("root dir", "*.ruleset", SearchOption.AllDirectories), Times.Once);
+            fileSystem.Verify(x => x.Directory.GetFiles("root dir", "*.props", SearchOption.AllDirectories), Times.Once);
+            fileSystem.Verify(x => x.Directory.GetFiles("root dir", "*.targets", SearchOption.AllDirectories), Times.Once);
+            fileSystem.Verify(x => x.Directory.GetFiles("root dir", "*.csproj", SearchOption.AllDirectories), Times.Once);
+            fileSystem.Verify(x => x.Directory.GetFiles("root dir", "*.vbproj", SearchOption.AllDirectories), Times.Once);
```

Comment:
If we have time, we might fetch the project files in a different way to make sure we only pick up the projects reference in the solution (the `RoslynProjectWalker` class gives us an easy way to find just the Roslyn projects).